--- openldap-2.4.4alpha/tests/data/slapd-config-undo.conf.dont-write-to-testdir	2007-02-13 22:22:33.000000000 +0200
+++ openldap-2.4.4alpha/tests/data/slapd-config-undo.conf	2007-04-23 12:45:51.000000000 +0200
@@ -20,4 +20,4 @@
 #monitor#database	monitor
 
 database config
-include		"configpw.conf"
+include		"@TESTDIR@/configpw.conf"
--- openldap-2.4.4alpha/tests/data/slapd-translucent-local.conf.dont-write-to-testdir	2007-02-13 22:22:33.000000000 +0200
+++ openldap-2.4.4alpha/tests/data/slapd-translucent-local.conf	2007-04-23 12:45:51.000000000 +0200
@@ -63,4 +63,4 @@
 acl-bind	binddn="uid=binder,o=translucent" credentials="bindtest"
 
 database config
-include		"configpw.conf"
+include		"@TESTDIR@/configpw.conf"
--- openldap-2.4.4alpha/tests/data/slapd-valsort.conf.dont-write-to-testdir	2007-02-13 22:22:33.000000000 +0200
+++ openldap-2.4.4alpha/tests/data/slapd-valsort.conf	2007-04-23 12:45:51.000000000 +0200
@@ -52,6 +52,6 @@
 valsort-attr            employeeType ou=users,o=valsort weighted alpha-ascend
 
 database config
-include configpw.conf
+include @TESTDIR@/configpw.conf
 
 #monitor#database	monitor
--- openldap-2.4.4alpha/tests/scripts/defines.sh.dont-write-to-testdir	2007-02-13 22:22:35.000000000 +0200
+++ openldap-2.4.4alpha/tests/scripts/defines.sh	2007-04-23 12:45:51.000000000 +0200
@@ -145,7 +145,7 @@
 SLAPADDLOG1=$TESTDIR/slapadd.1.log
 SLURPLOG=$TESTDIR/slurp.log
 
-CONFIGPWF=./configpw
+CONFIGPWF=$TESTDIR/configpw
 
 # args
 TOOLARGS="-x $LDAP_TOOLARGS"
@@ -169,9 +169,9 @@
 SLAPD="`pwd`/../servers/slapd/slapd -s0"
 SLURPD="`pwd`/../servers/slurpd/slurpd"
 LDAPPASSWD="$CLIENTDIR/ldappasswd $TOOLARGS"
-LDAPSASLSEARCH="$CLIENTDIR/ldapsearch $TOOLPROTO $LDAP_TOOLARGS -LLL"
-LDAPSEARCH="$CLIENTDIR/ldapsearch $TOOLPROTO $TOOLARGS -LLL"
-LDAPRSEARCH="$CLIENTDIR/ldapsearch $TOOLPROTO $TOOLARGS"
+LDAPSASLSEARCH="$CLIENTDIR/ldapsearch2.4 $TOOLPROTO $LDAP_TOOLARGS -LLL"
+LDAPSEARCH="$CLIENTDIR/ldapsearch2.4 $TOOLPROTO $TOOLARGS -LLL"
+LDAPRSEARCH="$CLIENTDIR/ldapsearch2.4 $TOOLPROTO $TOOLARGS"
 LDAPDELETE="$CLIENTDIR/ldapdelete $TOOLPROTO $TOOLARGS"
 LDAPMODIFY="$CLIENTDIR/ldapmodify $TOOLPROTO $TOOLARGS"
 LDAPADD="$CLIENTDIR/ldapmodify -a $TOOLPROTO $TOOLARGS"
--- openldap-2.4.4alpha/tests/scripts/test049-sync-config.dont-write-to-testdir	2007-02-13 22:22:36.000000000 +0200
+++ openldap-2.4.4alpha/tests/scripts/test049-sync-config	2007-04-23 12:45:51.000000000 +0200
@@ -43,7 +43,8 @@
 #
 
 echo "Starting producer slapd on TCP/IP port $PORT1..."
-$SLAPADD -F $CFPRO -n 0 -l $DYNAMICCONF
+sed -e "s,file\:\([^\/]+/\)*\([^\/]*\),file\:$TESTDIR\/\2,g" $DYNAMICCONF > ${DYNAMICCONF}.run
+$SLAPADD -F $CFPRO -n 1 -l ${DYNAMICCONF}.run
 cd $PRODIR
 $SLAPD -F ./slapd.d -h $URI1 -d $LVL $TIMING > ../$LOG1 2>&1 &
 PID=$!
--- openldap-2.4.4alpha/tests/scripts/test050-syncrepl-multimaster.dont-write-to-testdir	2007-02-13 22:22:36.000000000 +0200
+++ openldap-2.4.4alpha/tests/scripts/test050-syncrepl-multimaster	2007-04-23 12:45:51.000000000 +0200
@@ -42,6 +42,7 @@
 # - retrieve database over ldap and compare against expected results
 #
 
+cat $TESTDIR/configpw
 echo "Initializing server configurations..."
 $SLAPADD -F $CFCON -n 0 <<EOF
 dn: cn=config
@@ -52,7 +53,7 @@
 dn: olcDatabase={0}config,cn=config
 objectClass: olcDatabaseConfig
 olcDatabase: {0}config
-olcRootPW:< file:configpw
+olcRootPW:< file:$TESTDIR/configpw
 EOF
 
 $SLAPADD -F $CFPRO -n 0 <<EOF
@@ -64,7 +65,7 @@
 dn: olcDatabase={0}config,cn=config
 objectClass: olcDatabaseConfig
 olcDatabase: {0}config
-olcRootPW:< file:configpw
+olcRootPW:< file:$TESTDIR/configpw
 EOF
 
 echo "Starting producer slapd on TCP/IP port $PORT1..."
--- openldap-2.4.4alpha/tests/run.in.dont-write-to-testdir	2007-02-13 22:22:33.000000000 +0200
+++ openldap-2.4.4alpha/tests/run.in	2007-04-23 12:45:51.000000000 +0200
@@ -189,8 +189,9 @@
 # disable LDAP initialization
 LDAPNOINIT=true; export LDAPNOINIT

+mkdir -p $TESTDIR
 $SLAPPASSWD -g -n >$CONFIGPWF
-echo "rootpw `$SLAPPASSWD -T $CONFIGPWF`" >configpw.conf
+echo "rootpw `$SLAPPASSWD -T $CONFIGPWF`" >$TESTDIR/configpw.conf

 echo "Running ${SCRIPT}..."
 $SCRIPT $*
--- openldap-2.4.4alpha/tests/scripts/all.dont-write-to-testdir	2007-02-13 22:22:35.000000000 +0200
+++ openldap-2.4.4alpha/tests/scripts/all	2007-04-29 21:26:50.000000000 +0200
@@ -13,7 +13,8 @@
 ## top-level directory of the distribution or, alternatively, at
 ## <http://www.OpenLDAP.org/license.html>.
 
-SHTOOL="$SRCDIR/../build/shtool"
+. scripts/defines.sh
+echo $SHTOOL
 TB=`$SHTOOL echo -e "%B"`
 TN=`$SHTOOL echo -e "%b"`
 SLEEPTIME=10
@@ -23,10 +24,13 @@
 for CMD in $SRCDIR/scripts/test*; do
 	# remove cruft from prior test
 	if test $PRESERVE = yes ; then
-		/bin/rm -rf testrun/db.*
+		/bin/rm -rf $TESTDIR/db.*
 	else
-		/bin/rm -rf testrun
+		/bin/rm -rf $TESTDIR
 	fi
+	mkdir -p $TESTDIR
+	$SLAPPASSWD -g -n >$CONFIGPWF
+	echo "rootpw `$SLAPPASSWD -T $CONFIGPWF`" >$TESTDIR/configpw.conf
 
 	echo ">>>>> Starting ${TB}`basename $CMD`${TN} ..."
 	$CMD
